driver-async/src/main/com/mongodb/async/client/MongoCollectionImpl.java

```diff
@@ -934,48 +810,30 @@ public void createIndexes(final ClientSession clientSession, final List<IndexMod
 
     private void executeCreateIndexes(final ClientSession clientSession, final List<IndexModel> indexes,
                                       final CreateIndexOptions createIndexOptions, final SingleResultCallback<List<String>> callback) {
-        notNull("indexes", indexes);
-        notNull("createIndexOptions", createIndexOptions);
-
-        List<IndexRequest> indexRequests = new ArrayList<IndexRequest>(indexes.size());
-        for (IndexModel model : indexes) {
-            if (model == null) {
-                throw new IllegalArgumentException("indexes can not contain a null value");
-            }
-            indexRequests.add(new IndexRequest(toBsonDocument(model.getKeys()))
-                    .name(model.getOptions().getName())
-                    .background(model.getOptions().isBackground())
-                    .unique(model.getOptions().isUnique())
-                    .sparse(model.getOptions().isSparse())
-                    .expireAfter(model.getOptions().getExpireAfter(TimeUnit.SECONDS), TimeUnit.SECONDS)
-                    .version(model.getOptions().getVersion())
-                    .weights(toBsonDocument(model.getOptions().getWeights()))
-                    .defaultLanguage(model.getOptions().getDefaultLanguage())
-                    .languageOverride(model.getOptions().getLanguageOverride())
-                    .textVersion(model.getOptions().getTextVersion())
-                    .sphereVersion(model.getOptions().getSphereVersion())
-                    .bits(model.getOptions().getBits())
-                    .min(model.getOptions().getMin())
-                    .max(model.getOptions().getMax())
-                    .bucketSize(model.getOptions().getBucketSize())
-                    .storageEngine(toBsonDocument(model.getOptions().getStorageEngine()))
-                    .partialFilterExpression(toBsonDocument(model.getOptions().getPartialFilterExpression()))
-                    .collation(model.getOptions().getCollation()));
-        }
-        final CreateIndexesOperation createIndexesOperation = new CreateIndexesOperation(getNamespace(), indexRequests, writeConcern)
-                .maxTime(createIndexOptions.getMaxTime(MILLISECONDS), MILLISECONDS);
-        executor.execute(createIndexesOperation, clientSession, new SingleResultCallback<Void>() {
+        executor.execute(operations.createIndexes(indexes, createIndexOptions), clientSession, new SingleResultCallback<Void>() {
             @Override
             public void onResult(final Void result, final Throwable t) {
                 if (t != null) {
                     callback.onResult(null, t);
                 } else {
-                    callback.onResult(createIndexesOperation.getIndexNames(), null);
+                    callback.onResult(getIndexNames(indexes), null);
                 }
             }
         });
     }
 
+    private List<String> getIndexNames(final List<IndexModel> indexes) {
```

Comment:
Since CreateIndexesOperation is no longer visible here, getIndexNames had to move

Reply:
As this is duplicated with sync - it should be moved to the IndexHelper class

Reply:
I tried that, but then it requires pulling in the codec registry to convert Bson to BsonDocument.  I can to it though.

Reply:
Done